packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -949,12 +974,22 @@ class SliverAppBar extends StatefulWidget {
   /// If [snap] is true then a scroll that exposes the app bar will trigger an
   /// animation that slides the entire app bar into view. Similarly if a scroll
   /// dismisses the app bar, the animation will slide it completely out of view.
+  ///
```

Comment:
You might just repeat the link to two "floating"  animations here, one with floating on, and one with floating off, and the other properties static.  Same for the other two properties.

Reply:
I was thinking about that, but I think we'd have to repeat all 6 animations again because the interesting question I had most of the time is, "How does this property combine with pinned: true, how with pinned: false, etc"

Reply:
Repeating it is OK.  Often people get to the API docs via a link from an outside page, and we'd rather have duplicate information than make them click for it. I think if you just include the two on each one (one with the property in question off, and one on), then it shows more clearly what this specific property changes.  You can have the See Also still point to the SliverAppBar class docs for the comparison of all the combinations.

Thank goodness there's only three bools here... :-)

Reply:
Added. PTAL.